src/main/sensors/pitotmeter.c

```diff
@@ -242,11 +260,21 @@ STATIC_PROTOTHREAD(pitotThread)
             //
             // Therefore we shouldn't care about CAS/TAS and only calculate IAS since it's more indicative to the pilot and more useful in calculations
             // It also allows us to use pitot_scale to calibrate the dynamic pressure sensor scale
-            pitot.airSpeed = pitotmeterConfig()->pitot_scale * fast_fsqrtf(2.0f * fabsf(pitot.pressure - pitot.pressureZero) / SSL_AIR_DENSITY) * 100;
+
+            // NOTE ::filter pressure - apply filter when NOT calibrating for zero !!!
+            currentTimeUs = micros();
+            pitot.pressure = pt1FilterApply3(&pitot.lpfState, pitotPressureTmp, US2S(currentTimeUs - pitot.lastMeasurementUs));
+            pitot.lastMeasurementUs = currentTimeUs;
+
+            pitot.airSpeed = pitotmeterConfig()->pitot_scale * fast_fsqrtf(2.0f * fabsf(pitot.pressure - pitot.pressureZero) / SSL_AIR_DENSITY) * 100;  // cm/s
+            pitot.temperature = pitotTemperatureTmp;   // Kelvin
+
         } else {
+            pitot.pressure = pitotPressureTmp;
```

Comment:
why are we using the pressure at this point, when the calibration is not completed and we are also setting airspeed to 0?

Reply:
Before the ``if (pitotIsCalibrationComplete()) {`` (line 253) the pressure dP is kept in variable ``pitotPressureTmp``.
During normal operation ``pitot.pressure`` is initialized by the LPF function (line 266) which takes as an argument the ``pitotPressureTmp``.
During zero calibration which relies on ``pitot.pressure`` it is not initialized since LPF has been moved to normal operation part of ``if``. Hence the need for this line.